mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -453,19 +596,31 @@ LogicalResult MILARewritePattern::matchAndRewrite(linalg::GenericOp laGeneric,
                    << "Multiple generic inputs come from writeback\n");
         return failure();
       }
-      if (!inpIdxMap.isIdentity()) {
-        LLVM_DEBUG(llvm::dbgs() << "Writeback input not read with identity map "
-                                   "- earlier lowering should've fixed\n");
+      SmallVector<unsigned> broadcastedDims;
+      // This is not to allow broadcasting but due to canonical linalg
+      // form if the unit dims can carry affine expr to be zero in the
+      // translation, hence there is a following check.
+      if (!outToInMap.isMinorIdentityWithBroadcasting(&broadcastedDims)) {
```

Comment:
Ok, yeah, given the comment, this is the condition I'm interested in given your explanation above.
What we need is that the map output -> input be the identity, ideally, except that it some cases we'll get zeroes because of how broadcasts get squished down, so this is the condition we'd want.

My paranoia also says that we may also want to impose, in this particular case, the condition that
```
inv(op -> output) ; (op -> input) == inv(op -> input) ; op -> output
```
(where `;` is "and then"/composition), but I'm not sure the zeroes will work the same in both directions.

The underlying thing I want to make sure doesn't happen is the presence of non-trivial operations in the maps for the gemm result and the final output. So my gut is telling me that there's additional conditions on the maps for the writeback inputs and the final output.

That is, consider the following (which we might handle earlier but definitely should check for here)
```
%a = [gemm output]
%b = [elementwise add]
%c = [final result]
linalg.generic indexing_maps = [(d0, d1, d2) -> (d1, d2, d0), (d0, d1, d2) -> (d0, d1, d2), (d0, d1, d2) -> (d1, d2, d0)] ins(%a, %b) outs (%c) {
  bbN(%arg3, %arg4, %arg5) {
    %0 = add %arg3, %arg4
     linalg.yield %0
}
```

inv(output -> op) is `(d0, d1, d2) -> (d2, d0, d1)`, and so we have inv(output -> op) ; (op -> writeback) = identity

Similarly, looking at the bias tensor, the composition gives the map (d0, d1, d2) -> (d2, d0, d1)

Which ... might actually work, come to think of it

Because that says "to go from writeback coordinates to bias coordinates, permute (d0, d1, d2) -> (d2, d0, d1)"
which is equivalent to "to go from *bias* coordinates to *writeback* coordinates, permute the inverse of that, ie (d0, d1, d2) -> (d1, d2, d0)", which is ... correct.

But whether this'll work out correctly when all the maps are non-trivial permutations ... probably.